tests/unit-tests/internal/quiz-submission/submission/repositories/test-class-aggregate-submission-repository.php

```diff
@@ -0,0 +1,294 @@
+ */
+class Aggregate_Submission_Repository_Test extends \WP_UnitTestCase {
+	public function testCreate_UseTablesOn_CallsTablesBasedRepository(): void {
+		/* Arrange. */
+		$comments_based = $this->createMock( Comments_Based_Submission_Repository::class );
+		$tables_based   = $this->createMock( Tables_Based_Submission_Repository::class );
+		$repository     = new Aggregate_Submission_Repository( $comments_based, $tables_based, true );
+
+		/* Expect & Act. */
+		$tables_based
+			->expects( $this->once() )
+			->method( 'create' )
+			->with( 1, 2, 12.34 );
+
+		$repository->create( 1, 2, 12.34 );
+	}
+
+	public function testCreate_UseTablesOn_CallsCommentsBasedRepository(): void {
+		/* Arrange. */
+		$comments_based = $this->createMock( Comments_Based_Submission_Repository::class );
+		$tables_based   = $this->createMock( Tables_Based_Submission_Repository::class );
+		$repository     = new Aggregate_Submission_Repository( $comments_based, $tables_based, true );
+
+		/* Expect & Act. */
+		$comments_based
+			->expects( $this->once() )
+			->method( 'create' )
+			->with( 1, 2, 12.34 );
+
+		$repository->create( 1, 2, 12.34 );
+	}
+
+	public function testCreate_UseTablesOff_DoesntCallTablesBasedRepository(): void {
+		/* Arrange. */
+		$comments_based = $this->createMock( Comments_Based_Submission_Repository::class );
+		$tables_based   = $this->createMock( Tables_Based_Submission_Repository::class );
+		$repository     = new Aggregate_Submission_Repository( $comments_based, $tables_based, false );
+
+		/* Expect & Act. */
+		$tables_based
+			->expects( $this->never() )
+			->method( 'create' );
+
+		$repository->create( 1, 2, 12.34 );
+	}
+
+	public function testCreate_UseTablesOff_CallsCommentsBasedRepository(): void {
+		/* Arrange. */
+		$comments_based = $this->createMock( Comments_Based_Submission_Repository::class );
+		$tables_based   = $this->createMock( Tables_Based_Submission_Repository::class );
+		$repository     = new Aggregate_Submission_Repository( $comments_based, $tables_based, false );
+
+		/* Expect & Act. */
+		$comments_based
+			->expects( $this->once() )
+			->method( 'create' )
+			->with( 1, 2, 12.34 );
+
+		$repository->create( 1, 2, 12.34 );
+	}
+
+	public function testGetOrCreate_WhenHasSubmission_DoesntCallCreate(): void {
+		/* Arrange. */
+		$repository = $this->getMockBuilder( Aggregate_Submission_Repository::class )
+			->disableOriginalConstructor()
+			->setMethods( [ 'get', 'create' ] )
+			->getMock();
+
+		/* Expect & Act. */
+		$repository
+			->expects( $this->once() )
+			->method( 'get' )
+			->willReturn( $this->createMock( Submission::class ) );
+
+		$repository
+			->expects( $this->never() )
+			->method( 'create' );
+
+		$repository->get_or_create( 1, 2, 12.34 );
+	}
+
+	public function testGetOrCreate_WhenHasNoSubmission_CallsCreate(): void {
+		/* Arrange. */
+		$repository = $this->getMockBuilder( Aggregate_Submission_Repository::class )
```

Comment:
I was thinking a bit about this test. Using mocks for a unit under test, I consider as a dangerous practice that is okay to use as the last resort (for example, for abstract classes).
So, my next question was: how to test it then? The simplest answer was to set expectations for mocked comments/tables-based repositories.
WDYT?

Reply:
Yeah, this is a weird one and I wasn't sure how to go about it.

> set expectations for mocked comments/tables-based repositories.

Yep, that should work. 👍 I will make a separate PR for improving it.

I'm curious what the approach would be if didn't have the other repositories. Maybe make an integration test and check the side-effect of the `create` method? 🤔